api/routers/periodic_tasks.py

```diff
@@ -0,0 +1,107 @@
+from fastapi import APIRouter
+from pydantic import ValidationError
+from starlette import status
+from starlette.exceptions import HTTPException
+from typing import Optional
+
+from api.models.periodic_task import PeriodicTaskDTO, PeriodicTaskListDTO
+from api.utils import (
+    extract_config, handle_response, update_config, map_section_from_config, map_to_config_file_format)
+import logging
+logger = logging.getLogger(__name__)
+
+periodic_tasks_router = APIRouter()
+
+
+def get_periodic_tasks():
+    config = extract_config(config_type="periodic_tasks")
+    return [map_section_from_config(x, config) for x in config.keys()]
+
+
+@periodic_tasks_router.get("", response_model=PeriodicTaskListDTO,
+                           response_model_exclude_none=True)
+def list_periodic_tasks():
+    """
+        Returns the list of periodic tasks configured in the processor.
+    """
+    return {
+        "periodicTasks": get_periodic_tasks()
+    }
+
+
+@periodic_tasks_router.get("/{periodic_task_name}", response_model=PeriodicTaskDTO,
+                           response_model_exclude_none=True)
+def get_periodic_taskss(periodic_task_name: str):
+    """
+    Returns the configuration related to the periodic task <periodic_task_name>.
+    """
+    periodic_task = next((ps for ps in get_periodic_tasks() if ps["name"] == periodic_task_name), None)
+    if not periodic_task:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND,
+                            detail=f"The periodic_task: {periodic_task_name} does not exist")
+    return periodic_task
+
+
+@periodic_tasks_router.post("", response_model=PeriodicTaskDTO,
+                            status_code=status.HTTP_201_CREATED, response_model_exclude_none=True)
+async def create_periodic_task(new_periodic_task: PeriodicTaskDTO, reboot_processor: Optional[bool] = True):
+    """
+    Adds a periodic task.
+    """
+    if new_periodic_task.name != "reports":
+        raise ValidationError(f"Not supported periodic task named: {new_periodic_task.name}")
```

Comment:
Same than my last comment, these controllers are pretty complex, maybe it makes sense to move what logic we can to the models

Reply:
Are you talking about moving the name validation into the model or the whole logic?